cmake/FindFFTW3.cmake

```diff
@@ -38,7 +38,7 @@ find_package_handle_standard_args(FFTW3 DEFAULT_MSG FFTW3_LIBRARIES
 
 mark_as_advanced(FFTW3_LIBRARIES FFTW3_INCLUDE_DIR)
 
-if(FFTW3_FOUND)
+if(FFTW3_FOUND AND NOT TARGET FFTW3::FFTW3)
```

Comment:
It is an error to define the target if it already exists. this is common practice I think. Also walberla defines the target too IIRC.

Reply:
because walberla also imports the FFTW3 target, causing a collision:
```
CMake Error at cmake/FindFFTW3.cmake:42 (add_library):
  add_library cannot create imported target "FFTW3::FFTW3" because another
  target with the same name already exists.
Call Stack (most recent call first):
  build-seraue-release/_deps/walberla-src/CMakeLists.txt:963 (find_package)


CMake Error at cmake/FindFFTW3.cmake:44 (target_link_libraries):
  Cannot specify link libraries for target "FFTW3::FFTW3" which is not built
  by this project.
Call Stack (most recent call first):
  build-seraue-release/_deps/walberla-src/CMakeLists.txt:963 (find_package)
```

[cmake-developer(7)](https://cmake.org/cmake/help/v3.18/manual/cmake-developer.7.html?highlight=namespaced#a-sample-find-module)  says:
> When providing imported targets, these should be namespaced (hence the `Foo::` prefix); CMake will recognize that values passed to `target_link_libraries()` that contain `::` in their name are supposed to be imported targets (rather than just library names), and will produce appropriate diagnostic messages if that target does not exist (see policy CMP0028).
> ```cmake
> if(Foo_FOUND AND NOT TARGET Foo::Foo)
>   add_library(Foo::Foo UNKNOWN IMPORTED)
>   ...
> endif()
> ```

This is documented in the commit message 93d30a8e3d.

Reply:
sorry, I'm new to meaningful commit messages :smile: